base/set.jl

```diff
@@ -196,15 +196,26 @@ end
 Removes recurrences of items and returns the modified collection
 """
 function unique!(itr)
-	seen = Set{eltype(itr)}()
-	for (i, x) in enumerate(itr)
-		if x in seen
-			deleteat!(itr, i)
-		else
-			push!(seen, x)
-		end
-	end
-	return itr
+   T = eltype(itr)
```

Comment:
Do I need to make any changes to the method I have implemented ?
I will add test cases soon